File: Frontend/src/app/shared/enums/status-pagamento.enum.ts
export enum StatusPagamentoEnum {
  /**
    * Novo - Aguardando Geração de Lote
   */
    novo = 1,
    /**
    *  Lote Gerado - Aguardando Envio para o SAP
   */
    loteGerado = 2,
    /**
    *  Lote Cancelado
   */
    loteCancelado = 3,
     /**
    *  Lote Enviado - Aguardando Criação do Pedido SAP
   */
    loteEnviado = 4,
      /**
    *  Erro na Criação do Pedido SAP - Aguardando Criação de Lote
   */
    erroCriacaoSap = 5,
    /**
    * Pedido SAP Criado - Aguardando Recebimento Fiscal
     */
    pedidoSapCriado = 6,
    /**
    * Aguardando Envio para Cancelamento do Pedido SAP
     */
    aguardandoEnvioCancelamentoSap = 7,
     /**
    * Pedido SAP Enviado - Aguardando Cancelamento
     */
    pedidoSapEnviado = 8,
     /**
    * Erro no Cancelamento do Pedido SAP
     */
    erroCancelamentoSap = 9,
    /**
    * Pedido SAP Cancelado - Aguardando Geração de Lote
     */
    pedidoSapCancelado = 10,
    /**
    * Pedido SAP Pago
     */
    pedidoSapPago = 11,
    /**
    * Pedido SAP Pago Manualmente
     */
    pedidoSapPagoManualmente = 12,
    /**
    * Estorno
     */
    estorno = 13,
    /**
    * Sem Lançamento para o SAP
     */
    semLancamento = 14,
    /**
    * Pedido SAP Recebido Fiscal - Aguardando Confirmação de Pagamento
     */
    aguardandoConfirmacaoPagamento = 15,
    /**
    * Sem Lançamento para o SAP (Histórico)
     */
    semLancamentoHistorico = 17,
    /**
    * Lançamento de Controle
     */
    lancamentoControle = 18,
    /**
    * Pedido SAP Pago (lançamento retido RJ)
     */
     pedidoSapPagoLancamentoRetido = 19,
      /**
    * Pedido SAP Recebido Fiscal - Aguardando Confirmação de Pagamento (lançamento retido RJ)
     */
    aguardandoConfirmePagLancamentoRetido = 20,
     /**
    * Lote Automático Cancelado
     */
    loteAutomaticoCancelado = 21,
     /**
    * Lançamento Automático Cancelado
     */
    lancamentoAutomaticoCancelado = 22,
    /**
    * Pedido SAP Retido - RJ
     */
    pedidoSapRetido = 23,
}

File: Backend/Perlink.Oi.Juridico.Domain/SAP/Enum/StatusPagamentoEnum.cs
using System.ComponentModel;

namespace Perlink.Oi.Juridico.Domain.SAP
{
    public enum StatusPagamentoEnum
    {

        [Description("Novo - Aguardando Geracao de Lote")]
        Novo = 1,
        [Description("Lote Gerado - Aguardando Envio para o SAP")]
        LoteGeradoAguardandoEnvioSAP = 2,
        [Description("Lote Cancelado")]
        LoteCancelado = 3,
        [Description("Lote Enviado - Aguardando Criacao do Pedido SAP")]
        LoteEnviadoAguardandoCriacao = 4,
        [Description("Erro na Criacao do Pedido SAP - Aguardando Criação de Lote")]
        ErronaCriacaoPedidoSAP = 5,
        [Description("Pedido SAP Criado - Aguardando Recebimento Fiscal")]
        PedidoSAPCriadoAguardando = 6,
        [Description("Aguardando Envio para Cancelamento do Pedido SAP")]
        AguardandoEnvioCancelamento = 7,
        [Description("Pedido SAP Enviado - Aguardando Cancelamento")]
        PedidoSAPEnviadoAguardandoCancelamento = 8,
        [Description("Erro no Cancelamento do Pedido SAP")]
        ErroCancelamentoPedido = 9,
        [Description("Pedido SAP Cancelado - Aguardando Geração de Lote")]
        PedidoSapCancelado = 10,
        [Description("Pedido SAP Pago")]
        PedidoSAPPago = 11,
        [Description("Pedido SAP Pago Manualmente")]
        PedidoSAPPagoManualmente = 12,
        [Description("Estorno")]
        Estorno = 13,
        [Description("Sem Lancamento para o SAP")]
        SemLancamentoSAP = 14,
        [Description("Pedido SAP Recebido Fiscal - Aguardando Confirmação de Pagamento")]
        PedidoSAPRecebidoFiscal = 15,
        [Description("Sem Lancamento para o SAP (Historico)")]
        SemLancamentoSAPHistorico = 17,
        [Description("Lancamento de Controle")]
        LancamentoControle = 18,
        [Description("Pedido SAP Retido - RJ")]
        PedidoSAPRetido = 23,
        [Description("Lote Automatico Cancelado")]
        LoteAutomaticoCancelado = 21,
        [Description("Lancamento Automatico Cancelado")]
        LancamentoAutomaticoCancelado = 22,
        [Description("Pedido SAP Pago (lançamento retido RJ)")]
        PedidoSAPPagoLancamentoRetidoRJ = 19,
    }
}
